src/backend/parser/parse_relation.c

```diff
@@ -485,6 +485,8 @@ GetRTEByRangeTablePosn(ParseState *pstate,
 					   int varno,
 					   int sublevels_up)
 {
+	Insist(pstate);
```

Comment:
pstate may be NULL here https://github.com/arenadata/gpdb/blob/fd5939d4f491238bbd28be88b98949cd0e2d6fe0/src/backend/parser/parse_coerce.c#L404

Reply:
pstate is checked to be non-null before it's passed to `GetRTEByRangeTablePosn`, so I think we need this check. However, it can be replaced by assert, what do you think?

https://github.com/arenadata/gpdb/blob/0f972b1201828277979669744ab7bd007e623ea3/src/backend/parser/parse_coerce.c#L627-L630

Reply:
> pstate is checked to be non-null before it's passed to `GetRTEByRangeTablePosn`

What does the coerce_unknown_var function have to do with it? I was talking about the coerce_type function.

Reply:
And it's ok unless you point to a place where it's dereferenced
https://github.com/arenadata/gpdb/blob/fd5939d4f491238bbd28be88b98949cd0e2d6fe0/src/backend/parser/parse_coerce.c#L155

Reply:
Maybe we should have added a check here?
https://github.com/arenadata/gpdb/blob/fd5939d4f491238bbd28be88b98949cd0e2d6fe0/src/backend/parser/parse_coerce.c#L540-L546
```c
	if (inputTypeId == RECORDOID &&
		ISCOMPLEX(targetTypeId) &&
		PointerIsValid(pstate))
	{
		/* Coerce a RECORD to a specific complex type */
		return coerce_record_to_complex(pstate, node, targetTypeId,
										ccontext, cformat, location);
	}
```

Reply:
Changed condition to this, but used  != NULL to stick with exsisting style